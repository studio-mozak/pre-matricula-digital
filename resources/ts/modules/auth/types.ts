export interface Auth {
  level: number;
  name: string;
  schools: string[];
}
