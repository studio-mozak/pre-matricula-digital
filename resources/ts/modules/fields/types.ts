import { Nullable } from '@/types';

export type SetValues = (fields: Record<string, unknown>) => void;

export interface FieldPageOption {
  id?: string;
  name: Nullable<string>;
  weight: number;
}

export interface FieldPage {
  id: Nullable<string>;
  name: Nullable<string>;
  description: Nullable<string>;
  group: Nullable<'RESPONSIBLE' | 'STUDENT'>;
  type: Nullable<string>;
  internal: boolean;
  options: FieldPageOption[];
  deleteOptions?: string[];
  required?: boolean;
}

export interface ErrorResponse {
  message: string;
}
