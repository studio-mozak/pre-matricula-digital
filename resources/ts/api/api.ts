import axios, { AxiosInstance } from 'axios';

const token = window.config ? window.config.token : '';

export const http: AxiosInstance = axios.create({
  baseURL: String(import.meta.env.VITE_APP_API_URL || ''),
  headers: {
    Authorization: `Token ${token}`,
  },
});

axios.interceptors.response.use(
  (res) => {
    /**
     * If the response from the server has the "errors" object or the object "data" is empty,
     * here it is forced to go to the catch block automatically.
     */
    if (
      (res.data.errors || Object.keys(res.data).length === 0) &&
      !(res.data instanceof Blob)
    ) {
      return Promise.reject({
        response: {
          data: res.data,
        },
      });
    }

    return res;
  },
  (err) => Promise.reject(err)
);

export const rest = <T>(
  type: 'get' | 'post' | 'put' | 'delete',
  url: string,
  params?: object
): Promise<T> => http[type](url, params);

export const graphql = <T>(data: object, config?: object): Promise<T> =>
  axios.post('/graphql', data, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    ...config,
  });
