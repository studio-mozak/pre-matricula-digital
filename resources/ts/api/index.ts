export * from './api';
export * as City from './services/graphql/city';
export * as Resource from './services/graphql/resource';
