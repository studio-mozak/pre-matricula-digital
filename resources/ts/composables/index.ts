export * from './useLoader';
export * from './useLoaderAndShowErrors';
export * from './usePortal';
export * from './useModal';
export * from './useStudentProcessAndSuggestGrades';
