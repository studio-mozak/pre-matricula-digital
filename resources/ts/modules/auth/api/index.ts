/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as AuthRest from '@/modules/auth/api/services/rest/auth';
