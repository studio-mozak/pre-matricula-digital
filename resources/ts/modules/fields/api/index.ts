/**
 * Exporting the Axios Instance
 */
export { http, graphql } from '@/api';

/**
 * Exporting all modules
 */
export * as Field from './services/graphql/field';
