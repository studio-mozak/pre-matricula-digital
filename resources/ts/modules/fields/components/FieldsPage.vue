<template>
  <main>
    <h1>Campos</h1>
    <skeleton-form-list v-if="loadingData" class="mt-5" :items="20" />
    <div v-if="!loadingData" class="row mt-5">
      <div class="col-12 col-md-6">
        <div class="row">
          <div class="col-12">
            <p>Dados do(a) responsável pelo(a) aluno(a)</p>
            <hr />
          </div>
          <div
            v-for="responsibleField in responsible"
            :key="(responsibleField.id as string)"
            class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
            data-test="wrapper-checkbox"
            @click="openModalToUpdate(responsibleField)"
          >
            <label class="toggle-checkbox">
              <input type="checkbox" disabled :value="responsibleField.id" />
              <div
                :class="{ 'toggle-required': responsibleField.required }"
                class="toggle-text"
              >
                {{ responsibleField.name }}
              </div>
            </label>
          </div>
          <div v-if="responsible.length" class="col-12 mb-3">
            <label class="toggle-checkbox">
              <input type="checkbox" disabled />
              <div class="toggle-text toggle-required">Endereço</div>
            </label>
          </div>
          <div class="col-12">
            <x-btn
              data-test="btn-add-field-responsible"
              label="Adicionar novo campo"
              no-caps
              no-wrap
              color="primary"
              class="w-100"
              icon="mdi-plus"
              size="lg"
              @click="addField('RESPONSIBLE')"
            />
          </div>
        </div>
      </div>
      <div class="col-12 col-md-6 mt-5 mt-md-0">
        <div class="row">
          <div class="col-12">
            <p>Dados do(a) aluno(a)</p>
            <hr />
          </div>
          <div
            v-for="studentField in student"
            :key="(studentField.id as string)"
            class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
            @click="openModalToUpdate(studentField)"
          >
            <label class="toggle-checkbox">
              <input type="checkbox" disabled :value="studentField.id" />
              <div
                :class="{ 'toggle-required': studentField.required }"
                class="toggle-text"
              >
                {{ studentField.name }}
              </div>
            </label>
          </div>
          <div class="col-12">
            <x-btn
              data-test="btn-add-field-student"
              label="Adicionar novo campo"
              no-caps
              no-wrap
              color="primary"
              class="w-100"
              icon="mdi-plus"
              size="lg"
              @click="addField('STUDENT')"
            />
          </div>
        </div>
      </div>
    </div>
    <modal
      ref="fieldsModal"
      v-model="open"
      no-footer
      :title="(!!field.id ? 'Editar ' : 'Novo ') + 'Campo'"
    >
      <template #body>
        <x-form
          v-if="open"
          :loading="loading"
          :schema="schema"
          :initial-values="{ ...field }"
          @submit="saveField"
          @open:options="handleOpenOptions"
          @close:options="handleCloseOptions"
        >
          <template #default="{ values, errors, setValues }">
            <div
              v-if="
                values.type === 'SELECT' ||
                values.type === 'MULTI_SELECT' ||
                values.type === 'RADIO'
              "
              class="col-12"
            >
              <h3>Opções</h3>
              <div
                v-for="(item, i) in values.options"
                :key="i"
                class="d-flex justify-content-between"
              >
                <div class="w-100 mr-2">
                  <x-field
                    v-model="item.name"
                    :name="`options[${i}].name`"
                    container-class="form-group"
                    type="TEXT"
                    rules="required"
                    :errors="!!errors[`options[${i}].name`]"
                  />
                </div>
                <div class="mr-2">
                  <x-field
                    v-model="item.weight"
                    :name="`options[${i}].weight`"
                    container-class="form-group"
                    type="TEXT"
                    rules="required|numeric"
                    :errors="!!errors[`options[${i}].weight`]"
                  />
                </div>
                <div class="d-flex align-items-start">
                  <x-btn
                    v-tooltip.start-bottom="'Remover Opção'"
                    no-caps
                    no-wrap
                    class="w-100 h-75 flex-row text-danger border-danger"
                    icon="mdi-trash-can-outline"
                    @click="removeOption(values, i, setValues)"
                  />
                </div>
              </div>
              <x-btn
                no-caps
                no-wrap
                outline
                color="primary"
                class="w-100 mb-4"
                label="Adicionar opção"
                @click="addOption(values, setValues)"
              />
            </div>
          </template>
        </x-form>
      </template>
    </modal>
  </main>
</template>

<script setup lang="ts">
import { AppContext, computed, getCurrentInstance, onMounted, ref } from 'vue';
import {
  ErrorResponse,
  FieldPage,
  FieldPageOption,
  SetValues,
} from '@/modules/fields/types';
import { useLoader, useLoaderAndShowErrorByModal } from '@/composables';
import { Field } from '@/modules/fields/api';
import { FormSchema } from '@/types';
import Modal from '@/components/elements/Modal.vue';
import SkeletonFormList from '@/components/loaders/components/FormList.vue';
import XBtn from '@/components/elements/buttons/XBtn.vue';
import XField from '@/components/x-form/XField.vue';
import XForm from '@/components/x-form/XForm.vue';
import { fieldTypes } from '@/util';

const {
  loader: loaderData,
  data: fields,
  loading: loadingData,
} = useLoader<FieldPage[]>([]);

const loaderFieldError = ref();

const appContext = getCurrentInstance()?.appContext;

const { loader: loaderField, loading } = useLoaderAndShowErrorByModal<{
  errors?: ErrorResponse;
}>(appContext as AppContext, loaderFieldError);

const fieldsModal = ref<typeof Modal>();

const newOption = ref({
  name: null,
  weight: 0,
});

const newField = ref<FieldPage>({
  id: null,
  name: null,
  description: null,
  group: null,
  type: null,
  internal: false,
  options: [],
  deleteOptions: [],
});

const field = ref<FieldPage>({
  id: null,
  name: null,
  description: null,
  group: null,
  type: null,
  internal: false,
  options: [],
  deleteOptions: [],
});

const open = ref(false);
const types = ref(fieldTypes());

const responsible = computed(() =>
  fields.value.filter((f) => f.group === 'RESPONSIBLE')
);

const student = computed(() =>
  fields.value.filter((f) => f.group === 'STUDENT')
);

const schema = computed<FormSchema>(() => ({
  fields: [
    {
      label: 'Título da pergunta',
      name: 'name',
      type: 'TEXT',
      containerClass: 'col-12',
      rules: 'required',
    },
    {
      label: 'Descrição da pergunta',
      name: 'description',
      type: 'TEXT',
      containerClass: 'col-12',
      rules: '',
    },
    {
      label: 'Tipo da resposta',
      name: 'type',
      type: 'SELECT',
      options: types.value,
      containerClass: 'col-12',
      disabled: field.value.id ? true : undefined,
      rules: 'required',
      searchable: false,
    },
  ],
  buttons: [
    {
      type: 'submit',
      label: 'Salvar',
      class: 'btn btn-block btn-primary',
      containerClass: '',
      block: true,
      bind: {
        'data-test': 'btn-save-field',
      },
    },
  ],
  buttonsContainer: {
    class: '',
  },
}));

const addField = (group: 'RESPONSIBLE' | 'STUDENT') => {
  field.value = {
    ...newField.value,
    group,
    options: [],
    type: 'TEXT',
  };
  open.value = true;
};

const addOption = (values: FieldPage, setValues: SetValues) => {
  const { options } = values;

  let opts: FieldPageOption[] = [{ ...newOption.value }];

  if (options) {
    options.push({ ...newOption.value });
    opts = options;
  }

  setValues({
    ...values,
    options: opts,
  });
};

const openModalToUpdate = (item: FieldPage) => {
  field.value = { ...newField.value, ...item };
  open.value = true;
};

const createField = (model: FieldPage) => {
  loaderFieldError.value = {
    title: 'Erro ao criar o campo',
    description:
      'Aconteceu um erro ao criar o campo, por favor tente novamente em alguns instantes.',
  };
  loaderField(() => Field.post(model)).then(() => {
    open.value = false;
    loaderData(Field.list);
  });
};

const updateField = (model: FieldPage) => {
  loaderFieldError.value = {
    title: 'Erro ao editar o campo',
    description:
      'Aconteceu um erro ao editar o campo, por favor tente novamente em alguns instantes.',
  };
  loaderField(() => Field.put(model)).then(() => {
    open.value = false;
    loaderData(Field.list);
  });

  loading.value = true;
};

const saveField = (model: FieldPage) => {
  // Este trecho será reponsável por forçar o campo "weight" a ser um número
  if (model.options) {
    model.options.forEach((option) => {
      const opt = option;
      opt.weight = Number(option.weight);
    });
  }

  if (model.id) {
    updateField(model);
  } else {
    createField(model);
  }
};

const removeOption = (
  values: FieldPage,
  index: number,
  setValues: SetValues
) => {
  const option = values.options[index];
  const { options, deleteOptions } = values;

  options.splice(index, 1);
  if (option.id) {
    deleteOptions?.push(option.id);
  }

  setValues({
    ...values,
    options,
    deleteOptions,
  });
};

const handleOpenOptions = () => {
  if (field.value.options.length < 2 && fieldsModal.value) {
    fieldsModal.value?.setOverflowVisible(true);
  }
};

const handleCloseOptions = () => {
  if (fieldsModal.value) {
    fieldsModal.value?.setOverflowVisible(false);
  }
};

onMounted(() => loaderData(Field.list));
</script>
