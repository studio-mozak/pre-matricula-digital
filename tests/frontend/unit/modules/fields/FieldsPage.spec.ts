import { VueWrapper, flushPromises, mount } from '@vue/test-utils';
import { Field as FieldApi } from '@/modules/fields/api';
import { FieldPage } from '@/modules/fields/types';
import FieldsPage from '@/modules/fields/components/FieldsPage.vue';
import VTooltip from 'v-tooltip';
import { vi, expect } from 'vitest';
import { useModal } from '@/composables';

const dialogSpy = vi.fn();

vi.mock('@/composables/useModal');
vi.mock('@/modules/fields/api');

const fields: FieldPage[] = [
  {
    group: 'RESPONSIBLE',
    id: '2',
    internal: true,
    name: 'Data de nascimento',
    options: [],
    required: true,
    type: 'DATE',
  },
  {
    group: 'RESPONSIBLE',
    id: '3',
    internal: true,
    name: 'CPF',
    options: [],
    required: true,
    type: 'CPF',
  },
  {
    group: 'RESPONSIBLE',
    id: '7',
    internal: true,
    name: 'Naturalidade',
    options: [],
    required: false,
    type: 'CITY',
  },
  {
    group: 'STUDENT',
    id: '9',
    internal: true,
    name: 'Nome',
    options: [],
    required: true,
    type: 'TEXT',
  },
  {
    group: 'STUDENT',
    id: '10',
    internal: true,
    name: 'Data de nascimento',
    options: [],
    required: true,
    type: 'DATE',
  },
  {
    group: 'STUDENT',
    id: '13',
    internal: true,
    name: 'Certidão de nascimento',
    options: [],
    required: false,
    type: 'BIRTH_CERTIFICATE',
  },
];

const responsible = fields.filter((field) => field.group === 'RESPONSIBLE');

const student = fields.filter((field) => field.group === 'STUDENT');

const onMounted = async (payload: FieldPage[], errors?: boolean) => {
  if (!errors) {
    vi.mocked(FieldApi.list).mockResolvedValueOnce(payload);
  } else {
    vi.mocked(FieldApi.list).mockRejectedValueOnce({});
  }

  vi.mocked(useModal).mockImplementation(() => ({
    dialog: dialogSpy,
  }));

  const wrapper = mount(FieldsPage, {
    global: {
      directives: {
        tooltip: VTooltip,
      },
      stubs: ['x-field'],
    },
  });

  await flushPromises();

  await wrapper.vm.$nextTick();

  return wrapper;
};

describe('FieldsPage', () => {
  let wrapper: VueWrapper<InstanceType<typeof FieldsPage>>;

  beforeEach(async () => {
    wrapper = await onMounted(fields);

    await wrapper.vm.$nextTick();
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  test('the component exists', () => {
    expect(wrapper).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('trás os dados iniciais do servidor', () => {
    expect(wrapper.vm.fields).toMatchObject(fields);
    expect(wrapper.vm.responsible).toMatchObject(responsible);
    expect(wrapper.vm.student).toMatchObject(student);
  });

  test('deve listar os conteúdos dos campos de responsible e student', async () => {
    const contentList = wrapper.findAll('label.toggle-checkbox');

    // o valor de fields.length deve somar 1 pelo fato de que o campo de responsável
    // sempre terá um valor de endereço escrito de maneira hard coded, caso o responsible > 0

    expect(contentList.length).toEqual(fields.length + 1);
  });

  test('ao clicar no primeiro checkbox, deverá aparecer o modal para editar o campo', async () => {
    expect(wrapper.vm.open).toBeFalsy();

    const spy = vi.spyOn(wrapper.vm, 'openModalToUpdate');

    const wrapperCheckbox = wrapper.findAll(
      '[data-test="wrapper-checkbox"]'
    )[0];

    await wrapperCheckbox.trigger('click');

    expect(spy).toBeCalled();
    expect(wrapper.vm.open).toBeTruthy();
    expect(wrapper.vm.field).toEqual({
      ...wrapper.vm.newField,
      ...responsible[0],
    });
    expect(wrapper.find('.x-modal')).toBeTruthy();

    vi.restoreAllMocks();
  });

  test('não deve mostrar os campos se fields.length === 0', async () => {
    wrapper = await onMounted([], true);

    const contentList = wrapper.findAll('label.toggle-checkbox');

    expect(contentList.length).toEqual(0);

    wrapper.unmount();
    vi.restoreAllMocks();
  });

  describe('addField', () => {
    test('when I click on the button to add a new student field, addField function must be called', async () => {
      wrapper.vm.addField = vi.fn();

      const btn = wrapper.find('[data-test="btn-add-field-student"]');

      await btn.trigger('click');

      expect(wrapper.vm.addField).toBeCalledTimes(1);
      expect(wrapper.vm.addField).toBeCalledWith('STUDENT');
    });

    test('when I click on the button to add a new responsible field, addField function must be called', async () => {
      wrapper.vm.addField = vi.fn();

      const btn = wrapper.find('[data-test="btn-add-field-responsible"]');

      await btn.trigger('click');

      expect(wrapper.vm.addField).toBeCalledTimes(1);
      expect(wrapper.vm.addField).toBeCalledWith('RESPONSIBLE');
    });

    test('when addField is called with RESPONSIBLE param, modal variable must be true, and field variable must be filled', async () => {
      wrapper.vm.addField('RESPONSIBLE');

      await wrapper.vm.$nextTick();

      expect(wrapper.vm.open).toBeTruthy();
      expect(wrapper.vm.field).toMatchObject({
        id: null,
        name: null,
        internal: false,
        deleteOptions: [],
        group: 'RESPONSIBLE',
        options: [],
        type: 'TEXT',
      });
    });

    test('when addField is called with STUDENT param, modal variable must be true, and field variable must be filled', async () => {
      wrapper.vm.addField('STUDENT');

      await wrapper.vm.$nextTick();

      expect(wrapper.vm.open).toBeTruthy();
      expect(wrapper.vm.field).toMatchObject({
        id: null,
        name: null,
        internal: false,
        deleteOptions: [],
        group: 'STUDENT',
        options: [],
        type: 'TEXT',
      });
    });
  });

  describe('openModalToUpdate', () => {
    test('when openModalToUpdate is called, modal variable must be true', async () => {
      wrapper.vm.openModalToUpdate(responsible[0]);

      await wrapper.vm.$nextTick();

      expect(wrapper.vm.open).toBeTruthy();
      expect(wrapper.vm.field).toMatchObject({
        id: '2',
        internal: true,
        deleteOptions: [],
        group: 'RESPONSIBLE',
        options: [],
        name: 'Data de nascimento',
        required: true,
        type: 'DATE',
      });
    });
  });

  describe('createField', () => {
    test('when createField is called, it must call createField function', async () => {
      wrapper.vm.createField = vi.fn();

      wrapper.vm.createField();

      await wrapper.vm.$nextTick();

      expect(wrapper.vm.createField).toBeCalledTimes(1);
    });

    test('must finish the requisition with successfull', async () => {
      vi.mocked(FieldApi.post).mockResolvedValueOnce({});
      vi.mocked(FieldApi.list).mockResolvedValueOnce(fields);

      wrapper.vm.open = true;

      wrapper.vm.createField();

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(wrapper.vm.open).toBeFalsy();
    });

    test('must finish the requisition with error', async () => {
      vi.mocked(FieldApi.post).mockRejectedValueOnce({});

      wrapper.vm.open = true;

      wrapper.vm.createField();

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(wrapper.vm.open).toBeTruthy();
      expect(dialogSpy).toHaveBeenCalledWith({
        title: 'Erro ao criar o campo',
        description:
          'Aconteceu um erro ao criar o campo, por favor tente novamente em alguns instantes.',
        titleClass: 'danger',
        iconLeft: 'status-red',
      });
    });
  });

  describe('updateField', () => {
    test('when updateField is called, it must call updateField function', async () => {
      wrapper.vm.updateField = vi.fn();

      wrapper.vm.updateField();

      await wrapper.vm.$nextTick();

      expect(wrapper.vm.updateField).toBeCalledTimes(1);
    });

    test('must finish the requisition with successfull', async () => {
      vi.mocked(FieldApi.put).mockResolvedValueOnce({});
      vi.mocked(FieldApi.list).mockResolvedValueOnce(fields);

      wrapper.vm.open = true;

      wrapper.vm.updateField();

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(wrapper.vm.open).toBeFalsy();
    });

    test('must finish the requisition with error', async () => {
      vi.mocked(FieldApi.put).mockRejectedValueOnce({});

      wrapper.vm.open = true;

      wrapper.vm.updateField();

      await wrapper.vm.$nextTick();

      await flushPromises();

      expect(wrapper.vm.open).toBeTruthy();
      expect(dialogSpy).toHaveBeenCalledWith({
        title: 'Erro ao editar o campo',
        description:
          'Aconteceu um erro ao editar o campo, por favor tente novamente em alguns instantes.',
        titleClass: 'danger',
        iconLeft: 'status-red',
      });
    });
  });

  describe('saveField', () => {
    beforeEach(async () => {
      wrapper = await onMounted(fields);

      await wrapper.vm.$nextTick();
    });

    afterEach(() => {
      vi.clearAllMocks();
    });

    test('should exec createField function when field.id is null', async () => {
      vi.mocked(FieldApi.post).mockResolvedValueOnce({});
      vi.mocked(FieldApi.list).mockResolvedValueOnce(fields);

      wrapper.vm.saveField({
        ...responsible[0],
        id: null,
      });

      await wrapper.vm.$nextTick();

      await flushPromises();

      // if this expect assert to be true, it means that the createField function was called
      expect(wrapper.vm.open).toBeFalsy();
    });

    test('should exec updateField function when field.id is filled', async () => {
      vi.mocked(FieldApi.put).mockResolvedValueOnce({});
      vi.mocked(FieldApi.list).mockResolvedValueOnce(fields);

      wrapper.vm.saveField(responsible[0]);

      await wrapper.vm.$nextTick();

      await flushPromises();

      // if this expect assert to be true, it means that the updateField function was called
      expect(wrapper.vm.open).toBeFalsy();
    });
  });
});
