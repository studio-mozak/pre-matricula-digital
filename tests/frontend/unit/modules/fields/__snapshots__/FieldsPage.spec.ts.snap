// Vitest Snapshot v1

exports[`FieldsPage > should match snapshot 1`] = `
<main>
  <h1>
    Campos
  </h1>
  <!--v-if-->
  <div
    class="row mt-5"
  >
    <div
      class="col-12 col-md-6"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <p>
            Dados do(a) responsável pelo(a) aluno(a)
          </p>
          <hr />
        </div>
        
        <div
          class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
          data-test="wrapper-checkbox"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
              value="2"
            />
            <div
              class="toggle-required toggle-text"
            >
              Data de nascimento
            </div>
          </label>
        </div>
        <div
          class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
          data-test="wrapper-checkbox"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
              value="3"
            />
            <div
              class="toggle-required toggle-text"
            >
              CPF
            </div>
          </label>
        </div>
        <div
          class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
          data-test="wrapper-checkbox"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
              value="7"
            />
            <div
              class="toggle-text"
            >
              Naturalidade
            </div>
          </label>
        </div>
        
        <div
          class="col-12 mb-3"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
            />
            <div
              class="toggle-text toggle-required"
            >
              Endereço
            </div>
          </label>
        </div>
        <div
          class="col-12"
        >
          <button
            class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
            data-test="btn-add-field-responsible"
            role="button"
            style="font-size: 20px;"
            tabindex="0"
            type="button"
          >
            <span
              class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
            >
              <i
                aria-hidden="true"
                class="mdi mdi-plus pmd-icon on-left text-"
                role="img"
                style="margin-top: 1px;"
              >
                 
              </i>
              <span
                class="block"
              >
                Adicionar novo campo
              </span>
            </span>
            <transition-stub
              appear="false"
              css="true"
              name="x-transition--fade"
              persisted="false"
            >
              <!---->
            </transition-stub>
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-12 col-md-6 mt-5 mt-md-0"
    >
      <div
        class="row"
      >
        <div
          class="col-12"
        >
          <p>
            Dados do(a) aluno(a)
          </p>
          <hr />
        </div>
        
        <div
          class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
              value="9"
            />
            <div
              class="toggle-required toggle-text"
            >
              Nome
            </div>
          </label>
        </div>
        <div
          class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
              value="10"
            />
            <div
              class="toggle-required toggle-text"
            >
              Data de nascimento
            </div>
          </label>
        </div>
        <div
          class="col-12 col-sm-6 col-md-12 col-lg-6 mb-3"
        >
          <label
            class="toggle-checkbox"
          >
            <input
              disabled=""
              type="checkbox"
              value="13"
            />
            <div
              class="toggle-text"
            >
              Certidão de nascimento
            </div>
          </label>
        </div>
        
        <div
          class="col-12"
        >
          <button
            class="x-btn x-btn-item non-selectable no-outline x-btn--standard x-btn--rectangle bg-primary text-white x-btn--actionable pmd-focusable pmd-hoverable x-btn--no-uppercase w-100"
            data-test="btn-add-field-student"
            role="button"
            style="font-size: 20px;"
            tabindex="0"
            type="button"
          >
            <span
              class="x-btn__content text-center d-flex justify-content-center no-wrap text-no-wrap"
            >
              <i
                aria-hidden="true"
                class="mdi mdi-plus pmd-icon on-left text-"
                role="img"
                style="margin-top: 1px;"
              >
                 
              </i>
              <span
                class="block"
              >
                Adicionar novo campo
              </span>
            </span>
            <transition-stub
              appear="false"
              css="true"
              name="x-transition--fade"
              persisted="false"
            >
              <!---->
            </transition-stub>
          </button>
        </div>
      </div>
    </div>
  </div>
  <!--v-if-->
</main>
`;
